src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.cs

```diff
@@ -152,6 +153,14 @@ internal async Task<ImmutableArray<PragmaWarningCodeAction>> GetPragmaSuppressio
                     {
                         // global assembly-level suppress message attribute.
                         nestedActions.Add(new GlobalSuppressMessageCodeAction(suppressionTargetInfo.TargetSymbol, project, diagnostic, this));
+
+                        // local suppress message attribute
+                        // please note that in order to avoid issues with exising unit tests referencing the code fix
+                        // by their index this needs to be the last added to nestedActions
+                        if (suppressionTargetInfo.TargetMemberNode != null && suppressionTargetInfo.TargetSymbol.Kind != SymbolKind.Namespace)
+                        {
+                            nestedActions.Add(new LocalSuppressMessageCodeAction(this, suppressionTargetInfo.TargetSymbol, suppressionTargetInfo.TargetMemberNode, documentOpt, diagnostic));
```

Comment:
@mmihaly I am fine with you not providing FixAll support for this code action in this PR. However, I believe you need to add bail out code in `SuppressionFixAllProvider`. In it's current state, we will fail this contract assertion: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.FixAllProvider.cs,27

Reply:
It's actually not failing, but only by coincidence, as the `NestedSuppressionCodeAction.IsEquivalenceKeyForPragmaWarning` returns `true` for both local and pragma suppression (as they both start with "in Source"), see here:
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeFixes/Suppression/NestedSuppressionCodeAction.cs,25

That's the reason of pragama  suppression being performed when the fix all is chosen.

But I agree that this needs to be handled correctly, I just added the bail out code. I'm not sure this is the best way, but the result seems to be fine, in case the "Fix All" is chosen nothing happens, the window closes without any actions.

I'm working on the FixAll support, but that seems to be a bit more complex than I initially thought, and with my limited free time it might take a while to complete.